src/helpers/pivot/spreadsheet_pivot/spreadsheet_pivot.ts

```diff
@@ -0,0 +1,388 @@
+  }
+
+  getMeasure(name: string): PivotMeasure {
+    return this.definition.getMeasure(name);
+  }
+
+  getPivotHeaderValueAndFormat(domainStr: StringDomainArgs): FPayload {
+    const domain = toDomainArgs(domainStr);
+    const lastNode = domain.at(-1);
+    if (!lastNode) {
+      return { value: _t("Total") };
+    }
+    if (lastNode.field === "measure") {
+      return { value: this.getMeasure(lastNode.value).displayName };
+    }
+    const dimension = this.getDimension(lastNode.field);
+    const cells = this.filterDataEntriesFromDomain(this.dataEntries, domain);
+    const finalCell = cells[0]?.[dimension.nameWithGranularity];
+
+    if (!finalCell) {
+      return { value: "" };
+    }
+    if (finalCell.value === null) {
+      return { value: _t("(Undefined)") };
+    }
+    if (dimension.type === "date") {
+      if (dimension.granularity === "day") {
+        return {
+          value: toNumber(finalCell.value, this.getters.getLocale()),
+          format: this.getters.getLocale().dateFormat,
+        };
+      }
+      if (dimension.granularity === "month_number") {
+        return {
+          value: MONTHS[toNumber(finalCell.value, this.getters.getLocale())].toString(),
+        };
+      }
+    }
+    return {
+      value: finalCell.value,
+      format: finalCell.format,
+    };
+  }
+
+  getPivotCellValueAndFormat(measure: string, domainStr: StringDomainArgs): FPayload {
+    const domain = toDomainArgs(domainStr);
+    const dataEntries = this.filterDataEntriesFromDomain(this.dataEntries, domain);
+    if (dataEntries.length === 0) {
+      return { value: "" };
+    }
+    const values = dataEntries
+      .map((value) => value[measure])
+      .filter((cell) => cell && cell.type !== CellValueType.empty)
+      .filter(isDefined);
+    const aggregator = this.getMeasure(measure).aggregator || "count";
+    const operator = AGGREGATORS_FN[aggregator];
+    if (!operator) {
+      throw new Error(`Aggregator ${aggregator} does not exist`);
+    }
+    return {
+      value: values.length ? operator.fn([values], this.getters.getLocale()) : "",
+      format: operator.format(values[0]),
+    };
+  }
+
+  getPossibleFieldValues(groupBy: string): { value: string | number | boolean; label: string }[] {
+    //TODO This method should be implemented for the autocomplete feature
+    throw new Error("Method not implemented.");
+  }
+
+  getTableStructure(): SpreadsheetPivotTable {
+    if (!this.isValid()) {
+      throw new Error("Pivot is not valid !");
+    }
+    if (!this.table) {
+      this.table = dataEntriesToSpreadsheetPivotTable(this.dataEntries, this.definition);
+    }
+    return this.table;
+  }
+
+  getFields(): PivotFields | undefined {
+    return this.fields;
+  }
+
+  private filterDataEntriesFromDomain(dataEntries: DataEntries, domain: DomainArg[]) {
+    for (const node of domain) {
+      dataEntries = this.filterDataEntriesFromDomainNode(dataEntries, node);
+    }
+    return dataEntries;
```

Comment:
this function reassigns its parameter `dataEntries` for every loop. It's weird. Is it the intention ?

Reply:
maybe it could be re-written as 
```suggestion
   return   domain.reduce((current, acc) => this.filterDataEntriesFromDomainNode(acc, current) , dataEntries)
```